apps/gitness/src/pages-v2/pull-request/pull-request-conversation.tsx

```diff
@@ -227,33 +227,30 @@ export default function PullRequestConversationPage() {
   }, [sourceBranch, pullReqMetadata?.merged, pullReqMetadata?.closed])
 
   useEffect(() => {
-    if (branchError) {
-      if (pullReqMetadata?.merged || pullReqMetadata?.closed) {
-        setShowRestoreBranchButton(true)
-      } else {
-        setShowDeleteBranchButton(false)
-        createBranch({
-          repo_ref: repoRef,
-          body: {
-            name: pullReqMetadata?.source_branch || '',
-            target: pullReqMetadata?.source_sha,
-            bypass_rules: true,
-            dry_run_rules: true
-          }
-        }).then(res => {
-          if (res?.body?.rule_violations) {
-            const { checkIfBypassAllowed } = extractInfoFromRuleViolationArr(res.body?.rule_violations)
-            if (checkIfBypassAllowed) {
-              setShowRestoreBranchButton(true)
-            } else {
-              setShowRestoreBranchButton(false)
-            }
-          } else {
-            setShowRestoreBranchButton(true)
-          }
-        })
-      }
+    if (!branchError) return
+
+    if (pullReqMetadata?.merged || pullReqMetadata?.closed) {
+      return setShowRestoreBranchButton(true)
```

Comment:
Best to only return cleanup functions from a `useEffect`. This will be fine, but it's probably best to separate the call and the return since they aren't really connected.
```suggestion
      setShowRestoreBranchButton(true)
      return
```